ssl/ssl_sess.c

```diff
@@ -430,15 +430,6 @@ int ssl_get_new_session(SSL *s, int session)
             return 0;
         }
 
-        if (s->ext.hostname) {
-            ss->ext.hostname = OPENSSL_strdup(s->ext.hostname);
-            if (ss->ext.hostname == NULL) {
-                SSLfatal(s, SSL_AD_INTERNAL_ERROR, SSL_F_SSL_GET_NEW_SESSION,
-                         ERR_R_INTERNAL_ERROR);
-                SSL_SESSION_free(ss);
-                return 0;
-            }
-        }
```

Comment:
Why is this deleted?

Reply:
This is in ssl_get_new_session(), where we don't want to just mechanically copy over the value from the SSL to the SSL_SESSION -- in the new information model, that only happens when the SNI value is actually negotiated for the connection.  (I only noticed it was here because some tests weren't behaving as I expected, which shows the value of tests, I suppose.)